src/webapi/xml_http_request.rs

```diff
@@ -0,0 +1,129 @@
+use webapi::event_target::{IEventTarget, EventTarget};
+use webcore::value::{
+    Reference,
+    Value,
+};
+use webcore::try_from::TryInto;
+
+/// XMLHttpRequest object to make http requests
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)
+pub struct XMLHttpRequest( Reference );
+
+reference_boilerplate! {
+    XMLHttpRequest,
+    instanceof XMLHttpRequest
+    convertible to EventTarget
+}
+
+/// A number indicating the state of the `XMLHttpRequest`.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState)
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum ReadyState {
+    /// Client has been created. open() not called yet.
+    Unsent,
+    /// open() has been called.
+    Opened,
+    /// send() has been called, and headers and status are available.
+    HeadersReceived,
+    /// Downloading; responseText holds partial data.
+    Loading,
+    /// The operation is complete.
+    Done,
+}
+
+impl IEventTarget for XMLHttpRequest {}
+
+
+impl XMLHttpRequest {
+    /// Create new XMLHttpRequest
+    pub fn new() -> XMLHttpRequest {
+        XMLHttpRequest(js!{
+            return new XMLHttpRequest();
+        }.into_reference().unwrap())
+    }
+
+    /// Returns the current state of the request. 
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState)
+    pub fn ready_state(&self) -> ReadyState {
+        use self::ReadyState::*;
+        let state: u16 = js!( return @{self}.readyState; ).try_into().unwrap();
+        match state {
+            0 => Unsent,
+            1 => Opened,
+            2 => HeadersReceived,
+            3 => Loading,
+            4 => Done,
+            _ => unreachable!( "Unexpected value of XMLHttpRequest::readyState: {}", state )
+        }
+    }
+
+    /// Returns a DOMString that contains the response to the request as text, or None
+    /// if the request was unsuccessful or has not yet been sent.
+    ///
+    ///[(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseText)
+    pub fn response_text(&self) -> Option<String> {
+        let response = js!(return @{self}.responseText;);
+        match response {
+            Value::Null => None,
+            Value::String(resp) => Some(resp),
+            _ => None,
+        }
+    }
+
+    /// Returns an unsigned short with the status of the response of the request.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/status)
+    pub fn status(&self) -> u16 {
+        js!(return @{self}.status;).try_into().unwrap()
+    }
+
+    /// Open connection with given method (ie GET or POST), and the url to hit
+    /// Async is set for now because it's more ergonimic in rust and it is good form in javascript
```

Comment:
"Async is set for now because it's more ergonimic in rust and it is good form in javascript" -> this seems unnecessary. You can document that it's asynchronous, but it's not necessary to explain it like this.